src/Parsers/IParser.h

```diff
@@ -38,8 +38,11 @@ struct Expected
             return;
         }
 
-        if ((current_pos == max_parsed_pos) && (find(variants.begin(), variants.end(), description) == variants.end()))
-            variants.push_back(description);
+        if (current_pos == max_parsed_pos)
+        {
+            if ((current_pos == max_parsed_pos) && (std::find(variants.begin(), variants.end(), description) == variants.end()))
+                variants.push_back(description);
+        }
```

Comment:
I feel like `current_pos == max_parsed_pos` is now checked twice
```suggestion
        if ((current_pos == max_parsed_pos) && (std::find(variants.begin(), variants.end(), description) == variants.end()))
            variants.push_back(description);
```

Reply:
Yes, it was a mistake when reverting the changes. I'll fix it